core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -201,8 +405,206 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
+        Map<String, Float> fields = new HashMap<>();
+        for (int i = 0; i < size; i++) {
+            String field = in.readString();
+            Float weight = in.readFloat();
+            fields.put(field, weight);
+        }
+        result.fields(fields);
+
+        result.flags(in.readInt());
+        result.analyzer(in.readOptionalString());
+
+        result.defaultOperator(Operator.parseFromInt(in.readInt()));
+        result.lowercaseExpandedTerms(in.readBoolean());
+        result.lenient(in.readBoolean());
+        result.analyzeWildcard(in.readBoolean());
+
+        String localeStr = in.readString();
+        result.locale(Locale.forLanguageTag(localeStr));
+
+        result.queryName(in.readOptionalString());
+        result.minimumShouldMatch(in.readOptionalString());
+
+        return result;
+    }
+
+    /**
+     * Writes the current {@link SimpleQueryStringBuilder} object to the given
+     * output stream.
+     * 
+     * @param out The stream to write the object to.
+     **/
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(queryText);
+        out.writeFloat(boost);
+        out.writeInt(fieldsAndWeights.size());
+        for (Map.Entry<String, Float> entry : fieldsAndWeights.entrySet()) {
+            out.writeString(entry.getKey());
+            out.writeFloat(entry.getValue());
+        }
+        out.writeInt(flags);
+        out.writeOptionalString(analyzer);
+        out.writeInt(defaultOperator.ordinal());
+        out.writeBoolean(settings.lowercaseExpandedTerms());
+        out.writeBoolean(settings.lenient());
+        out.writeBoolean(settings.analyzeWildcard());
+        out.writeString(settings.locale().toLanguageTag());
+
+        out.writeOptionalString(queryName);
+        out.writeOptionalString(minimumShouldMatch);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int hashCode() {
+        return Objects.hash(fieldsAndWeights, analyzer, defaultOperator, queryText, queryName, minimumShouldMatch, settings);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        SimpleQueryStringBuilder other = (SimpleQueryStringBuilder) obj;
+        return Objects.equals(fieldsAndWeights, other.fieldsAndWeights) && Objects.equals(analyzer, other.analyzer)
+                && Objects.equals(defaultOperator, other.defaultOperator) && Objects.equals(queryText, other.queryText)
+                && Objects.equals(queryName, other.queryName) && Objects.equals(minimumShouldMatch, other.minimumShouldMatch)
+                && Objects.equals(settings, other.settings);
+    }
+
+    /**
+     * Class encapsulating the settings for the SimpleQueryString query, with
+     * their default values
+     */
+    public static class Settings {
```

Comment:
thanks for you explanation, I understand more now. I totally missed that that SimpleQueryParser != SimplerQueryStringParser... Settings should have been package private from day one ideally as it's only internal, used to pass parameters from SimpleQueryStringParser to SimpleQueryParser. With this refactoring though we add the query builder in-between effectively: SimplerQueryStringParser => SimpleQueryStringBuilder => SimpleQueryParser. I agree on keeping the class, I wonder again where the class belongs (builder or internal parser), either way I think the class shouldn't be public and the public constants should be moved to the top class. Java api users shouldn't get to see this implementation detail.